project-koku/koku
# [COST-3047] - update product-code trino sql

## Jira Ticket

[COST-3047](https://issues.redhat.com/browse/COST-3047)

## Description

- update product names for AWS-Marketplace entries
- update resource matching for OCP-on-AWS to match, e.g. 
```
aws:  arn:aws:ec2:us-east-1a:9999999999999:instance/i-25555555
ocp: i-25555555
```

## Testing

1. on main
1. Using the attached yml files (convert them to yamls). Replace the `dev/scripts/nise_ymls/ocp_on_aws` files.
2. `make create-test-customer; make load-test-customer-data test_source=AWS`
3. hit [http://localhost:8000/api/cost-management/v1/reports/aws/costs/?group_by[service]=*](http://localhost:8000/api/cost-management/v1/reports/aws/costs/?group_by[service]=*) and see:
```
            "services": [
                {
                    "service": "5hnnev4d0v7mapf09j0v8of0o2",
                    "values": [
                        {
                            "date": "2022-10-16",
                            "service": "5hnnev4d0v7mapf09j0v8of0o2",

```
4. hit [http://localhost:8000/api/cost-management/v1/reports/openshift/infrastructures/aws/costs/?group_by[service]=*](http://localhost:8000/api/cost-management/v1/reports/openshift/infrastructures/aws/costs/?group_by[service]=*) and see no data:
```
    "data": [
        {
            "date": "2022-10-16",
            "services": []
        },
```
5. delete all the data: `make delete-test-customer-data; make clear-testing; make clear-trino-data`
7. checkout branch
8. re-ingest all the data: `make create-test-customer; make load-test-customer-data test_source=AWS`
9. recheck [http://localhost:8000/api/cost-management/v1/reports/aws/costs/?group_by[service]=*](http://localhost:8000/api/cost-management/v1/reports/aws/costs/?group_by[service]=*) and see `5hnnev4d0v7mapf09j0v8of0o2` replaced with `Red Hat OpenShift Service on AWS`:
```
            "services": [
                {
                    "service": "Red Hat OpenShift Service on AWS",
                    "values": [
                        {
                            "date": "2022-10-16",
                            "service": "Red Hat OpenShift Service on AWS",
```
10. recheck [http://localhost:8000/api/cost-management/v1/reports/openshift/infrastructures/aws/costs/?group_by[service]=*](http://localhost:8000/api/cost-management/v1/reports/openshift/infrastructures/aws/costs/?group_by[service]=*) and see data and the `Red Hat OpenShift Service on AWS` service:
```
            "services": [
                {
                    "service": "Red Hat OpenShift Service on AWS",
                    "values": [
                        {
                            "date": "2022-10-16",
                            "service": "Red Hat OpenShift Service on AWS",
                            "clusters": [
                                "Test OCP on AWS"
                            ],

```

## Notes

...
[aws_static_data.yml.txt](https://github.com/project-koku/koku/files/9863229/aws_static_data.yml.txt)
[ocp_static_data.yml.txt](https://github.com/project-koku/koku/files/9863231/ocp_static_data.yml.txt)
